### DISCUSSION TITLE
Checking bead packing

### DISCUSSION THREAD

**Discussion by svihuri** [Vattulainen-Group/CellSimGmx#4] [General]
Hi everyone, 

We just discussed with Shreyas that the code should probably check that the membrane beads are not packed too far or too close from each other. Few questions for the implementation:

1. Is sigma (0.47) from the [ nonbond_params ] section of the force field file the correct parameter to represent bead radius? 
2. Would 2*sigma +- 0.2 nm be an okay range? 
3. If the user asks for e.g. too many beads for a given cell radius, should the program throw an error and exit?

**Comment by molecularmachinist** [comment]
A useful "epsilon" to add between the two beads would be 10% of the average sigma of the two beads. That way they will always be separated significantly without a hard number.

**Comment by svihuri** [comment]
I just tested a one cell system and with input 180 particles and cell radius 1.85, the cell looks like this after energy minimisation:
![em_cell](https://github.com/Vattulainen-Group/CellSimGmx/assets/75125404/da4fe278-42b6-4f49-929e-479f5a8eb700)

Almost all of the particles are closer than sigma+0.1*sigma (just to test) after initial packing using Golden spiral:
Creating a system with 1 cells and 1 layers
Minimum separation between particles: 0.517
Particles 1 and 4 are too close: 0.4408789722953156
Particles 1 and 6 are too close: 0.4680637670247025
Particles 2 and 5 are too close: 0.48679529091059476
Particles 2 and 7 are too close: 0.45221767277589076
Particles 3 and 8 are too close: 0.4548042153034793
Particles 3 and 11 are too close: 0.49822806701037986
...

**Comment by molecularmachinist** [comment]
So that would imply packing is not the issue unless they are too tightly packed. In that case the initial packing will generate high forces that might lead to large deviations from the starting geometry.
